android/app/src/main/java/app/organicmaps/widget/placepage/PlacePageUtils.java

```diff
@@ -79,7 +80,11 @@ static String toString(@BottomSheetBehavior.State int state)
   public static void copyToClipboard(Context context, View frame, String text)
   {
     Utils.copyTextToClipboard(context, text);
-    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
+
+    KeyguardManager keyguardManager = (KeyguardManager) context.getSystemService(Context.KEYGUARD_SERVICE);
+
+    //only show snackBar if device is locked or android sdk is earlier than TIRAMISU.
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU || Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU && keyguardManager.isDeviceLocked())
```

Comment:
isDeviceLocked available starting from LOLLIPOP_MR1 (API 22), while Organic Maps works from LOLLIPOP (API 21). Why is TIRAMISU (API 33) check needed here?

Reply:
the snack bar should be shown only on lockscreen for android version 13 and higher, when unlocked the preview popup is shown by default.

for android version lower than 13, the snackbar should be shown regardless of device is locked or not.

The if condition needs fixing, working on it.

Reply:
how about this modification: 
``` if(Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU)
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      }
      else if (keyguardManager.isDeviceLocked())
      {
        Utils.showSnackbarAbove(view.getRootView().findViewById(R.id.pp_buttons_layout), view,
                                ctx.getString(R.string.copied_to_clipboard, phoneNumber));
      } ```